lib/ruby_lsp/document.rb

```diff
@@ -177,6 +177,16 @@ def locate(node, char_position, node_types: [])
       [closest, parent, nesting.map { |n| n.constant_path.location.slice }]
     end
 
+    sig { returns(T::Boolean) }
+    def sigil_is_true_or_higher?
+      [/# typed: true/, /# typed: strict/, /# typed: strong/].any? { |pattern| pattern.match?(@source) }
```

Comment:
This is quite a weak check and historically we've seen instances where this resulted in a false-positive. For example, a test file at `typed: false` that has some test Ruby code string in it which includes `# typed: true` would check as having sigil true or higher according to this check:
```ruby
# typed: false

def foo
  <<~CODE
    # typed: true

    def something; end
  CODE
end
```